### PR TITLE
add Column macro and update version

### PR DESCRIPTION
**Pull request by bossinc**
I am not sure if there is a better way to make a frontend macro when we need the results of the adhoc filter.

### CODE REVIEW

File: src/data/CHDatasource.ts [hunk 1]
```diff
@@ -66,6 +67,13 @@ export class Datasource extends DataSourceWithBackend<CHQuery, CHConfig> {
     };
   }
 
+  private addColumnTemplateVariable(scoped: ScopedVars, adHocFilters: any): ScopedVars {
+    if (!adHocFilters || !scoped) return scoped;
+    const col = adHocFilters.length > 0 ? adHocFilters[0].key : '';
+    scoped['__column'] = { text: col, value: col };
```
Comment (scottlepp) [inline src/data/CHDatasource.ts:73]: probably shouldn't mutate scoped here rather:
```
return {...scoped, '__column': { text: col, value: col }}
```

File: src/data/CHDatasource.ts [hunk 2]
```diff
@@ -66,6 +67,13 @@ export class Datasource extends DataSourceWithBackend<CHQuery, CHConfig> {
     };
   }
 
+  private addColumnTemplateVariable(scoped: ScopedVars, adHocFilters: any): ScopedVars {
+    if (!adHocFilters || !scoped) return scoped;
+    const col = adHocFilters.length > 0 ? adHocFilters[0].key : '';
```
Comment (scottlepp) [inline src/data/CHDatasource.ts:72]: hmm not sure if having multiple adhoc filters is a valid scenario, but if so this could be a problem

**Review by scottlepp — CHANGES_REQUESTED** [review]

**Review by scottlepp — COMMENTED** [review]
Perhaps a bit more context/description as to why we need this....or the issue this relates to.

File: src/data/CHDatasource.ts (hunk 2 above)
Comment (bossinc) [reply src/data/CHDatasource.ts:72]: Oh you are totally right. I didnt think about having an adhoc filter like this `SELECT DISTINCT machine_name, machine_group FROM mgbench.logs1`

**Comment by bossinc** [comment]
> Perhaps a bit more context/description as to why we need this....or the issue this relates to.

https://github.com/grafana/clickhouse-datasource/pull/135#issuecomment-1161703207

**Pull request closed without merging**